### PR TITLE
feat: add html to pdf ci application

### PR DESCRIPTION
**Pull request by TGiles**
This adds v0.0.1 of the html-to-pdf-ci tool.

This tool allows a user to convert HTML documents into PDFs as part of a CI workflow

### CODE REVIEW

**Comment by TGiles** [comment]
Any feedback is appreciated!

File: README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+The destination should exist before generating the PDF!
+
+### Source path prompt
+
+The path to your HTML source document.
+The path must end with the actual HTML document, e.g. `somePath/index.html`.
+
+### Site path prompt
+
+The location of your generated static site.
```
Comment (dylansheffer) [inline README.md:51]: What is this parameter used for?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+In most static site generators, this location is called `_site` but please pay attention if you have changed the generated directory name!
```
Comment (dylansheffer) [inline README.md:52]: I would change this language to be more static site generator agnostic. `_site` can be the default, but this sentence should be phrased a bit differently. Maybe something like this.

"The default location is `_site`. However, if your static site generator uses a different folder name or you modified the output directory's name, use that directory instead"

File: README.md (hunk 1 above)
Comment (dylansheffer) [reply README.md:51]: Do you have to use a static site generator for this tool? My resume is just plain html and css

File: cli.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+inquirer.registerPrompt('fuzzypath', require('inquirer-fuzzy-path'));
+
+const fs = require('fs');
+const path = require('path');
+const htmlToPdfCI = require('./index');
+const package = require('./package.json');
+const ci = process.argv[2];
+let configFile = {};
+try {
+    configFile = JSON.parse(fs.readFileSync(path.join(process.cwd(), '.htmltopdf.json')));
```
Comment (dylansheffer) [inline cli.js:12]: What does `configFile` equal when fs fails and then it is parsed? Would it make sense to manually set configFile back to `{}` in the catch?

Comment (dylansheffer) [reply cli.js:12]: Maybe also log that there's no config file?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+const path = require('path');
+const puppeteer = require('puppeteer');
+const express = require('express');
+
+const main = async (answers) => {
+    try {
+        const port = 5001;
+        const destPath = path.join(process.cwd(), answers.destinationPath);
+        const fullResumePath = `http://localhost:${port}/${answers.sourcePath}`;
```
Comment (dylansheffer) [inline index.js:9]: variable should be renamed to `fullSourcePath`. This won't just be used for resumes

File: index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        const app = express();
+        app.use(express.static(path.join(process.cwd())));
+        const server = app.listen(port, async () => {
+            const browser = await puppeteer.launch();
+            const page = await browser.newPage();
+            if (!destFilename.endsWith('.pdf')) {
+                destFilename += '.pdf';
+            }
+            await page.goto(fullResumePath, { waitUntil: 'networkidle0' });
+            await page.pdf({
```
Comment (dylansheffer) [inline index.js:20]: Would there be a way to configure custom arguments for this in the .html-to-pdf.json file? My particular resume uses very tiny margins in order to print correctly. I would also assume that other people would have different needs as well (different paper sizes, landscape, etc)

File: cli.js (hunk 3 above)
Comment (TGiles) [reply cli.js:12]: If `fs` fails to find the file, it throws an error. So in that case, the missing file would never be parsed.

File: README.md (hunk 1 above)
Comment (TGiles) [reply README.md:51]: Good catch. This parameter is not needed anymore. Will be removed.

File: index.js (hunk 5 above)
Comment (TGiles) [reply index.js:20]: Added prompts and logic to handle:
* Margins
* Landscape vs Portrait
* Paper format (Letter, A4, etc)
* Print background graphics

File: index.js (hunk 4 above)
Comment (TGiles) [reply index.js]: Updated

**Pull request merged**